usaspending_api/common/experimental_api_flags.py

```diff
@@ -17,4 +25,46 @@ def is_experimental_elasticsearch_api(request: Request) -> bool:
     """
     Returns True or False depending on if the expected_header_value matches what is sent with the request.
     """
-    return request.META.get(EXPERIMENTAL_API_HEADER) == ELASTICSEARCH_HEADER_VALUE
+    use_es = request.META.get(EXPERIMENTAL_API_HEADER) == ELASTICSEARCH_HEADER_VALUE
+    if not use_es:
+        mirror_request_to_elasticsearch(request)
+    else:
+        logger.info(
+            f"Found {EXPERIMENTAL_API_HEADER} request header. "
+            f"Will use Elasticsearch for backing store where implemented"
+        )
+    return use_es
+
+
+def mirror_request_to_elasticsearch(request: Union[HttpRequest, Request]):
+    """ Duplicate request and send-again against this server, with the ES header attached to mirror
+        non-elasticsearch load against elasticsearch for load testing
+    """
+    url = request.build_absolute_uri()
+    data = json.dumps(request.data)
+    headers = {
+        **request.headers,
+        HttpHeaders.parse_header_name(EXPERIMENTAL_API_HEADER): ELASTICSEARCH_HEADER_VALUE,
+    }
+
+    logger.warning(
+        f"Mirroring inbound request with elasticsearch experimental header.\n"
+        f"\tOriginial Request details: {request._request}\n"
+        f"\tMirrored Request details: \n\t\turl = {url}, \n\t\tdata = {data}, \n\t\theaders = {headers}"
+    )
+
+    # NOTE: Purposely desiring an immediate timeout, with ignoring of that timeout error,
+    # since this is a fire-and-forget way to siphon off duplicate load-testing traffic to the server,
+    # without disrupting the primary request
+    try:
+        if request.method == "GET":
+            requests.get(url, data, headers=headers, timeout=0.01)
+        else:
```

Comment:
What about "OPTIONS" requests? Do we do an "UPDATE" or "INSERT" anywhere? I doubt it, but it might be safer to ignore anything other than GET and POST in order to avoid giving a bunch of 405s everywhere.

Reply:
I concur. Safer to explicitly check `POST` with `elif` and then just do nothing in all other cases.